libs/extensions/std/exec/src/archive-interpreter-executor.ts

```diff
@@ -133,36 +100,47 @@ export class ArchiveInterpreterExecutor extends AbstractBlockExecutor<
       )) {
         if (!archivedObject.dir) {
           const content = await archivedObject.async('arraybuffer');
-          // Ext incl. leading dot
-          const extName = path.extname(archivedObject.name);
-          const fileName = path.basename(archivedObject.name);
-          const mimeType =
-            inferMimeTypeFromContentTypeString(extName) ||
-            MimeType.APPLICATION_OCTET_STREAM;
-          const fileExtension =
-            inferFileExtensionFromFileExtensionString(extName) ||
-            FileExtension.NONE;
-          const file = new BinaryFile(
-            fileName,
-            fileExtension,
-            mimeType,
-            content,
-          );
-          const addedFile = fs.putFile(
-            InMemoryFileSystem.getPathSeparator() + relPath,
-            file,
-          );
-          assert(addedFile != null);
+          this.createBinaryAndPutFile(fs, archivedObject.name, content, {
+            relPath: relPath,
+          });
         }
       }
       return R.ok(fs);
     } catch (error: unknown) {
-      return R.err({
-        message: `Unexpected Error ${
-          error instanceof Error ? error.message : JSON.stringify(err)
-        } occured during processing`,
-        diagnostic: { node: context.getCurrentNode(), property: 'name' },
-      });
+      return R.err(this.generateErrorObject(context, error));
     }
   }
+
+  private createBinaryAndPutFile(
```

Comment:
I think there are two problems with this:

- Methods that have an "and" in their name are almost always to large and can be split up in an obvious way ;). In this case, I'd split this into `createBinary` and `putFile` theoretically.
- Changing objects that are parameters to a function as a sideeffect is often confusing and leads to errors. Here I would not change the inmemoryfilesystem as sideeffect of the function.

Combined, I'd rename the method to `createFileFromArchive` (or similar) and just have line 120 to 128 in it, return the new `BinaryFile`. Then habe the fs.putfile and assert outside of the function on it's own, I don't think they need their own function.

Reply:
- Moved `fs.putFile` and assertion outside of function. 
- Renamed function to `createFileFromArchive` and have it return the new `BinaryFile`. 
- Removed now unused parameters.